File: Back-end/BusinessLayer/Services/StatisticService.cs
namespace BusinessLayer.Services;

using AutoMapper;
using Azure;
using Contracts;
using DataLayer.Entities;
using DataLayer.Repository.Interface;
using Interfaces;
using Microsoft.EntityFrameworkCore;
using Models;
using System;
using System.Threading.Tasks;

public class StatisticService(IStatisticRepository statisticRepository, IMapper mapper) : IStatisticService
{

    public async Task<GetStatisticResponse> GetStatistic(Guid spaceId, int year)
    {
        var response = new GetStatisticResponse();

        var result = await statisticRepository.GetReservationDates(spaceId, year);

        response.Statistic = mapper.Map<List<StatisticDTO>>(result);

        return response;
    }
}

File: Front-end/BusinessLayer/Services/StatisticService.js
import { StatisticRepository } from "../../DataLayer/Repository/StatisticRepository.js";
import { StatisticMapper } from "../../BusinessLayer/Mappers/StatisticMapper.js";
import { displayPrompt } from "../../WebLayer/utilities/Prompt.js";
import { StatisticDTO } from "../Models/StatisticDTO.js";

export class StatisticService {
  constructor() {
    this.statisticRepositpry = new StatisticRepository();
    this.statisticMapper = new StatisticMapper();
  }


  async getStatistic(spaceid, year) {
    const response = await this.statisticRepositpry.getStatistic(spaceid, year);
    if (response.error.hasError) {
      displayPrompt(".prompt-save", response.error.errorMessage, false);
      return;
    }

    var statisiticResult = this.statisticMapper.MapStatistic(response.statistic);

    return new StatisticDTO(statisiticResult, response.invoiceStatus.paid, response.invoiceStatus.unpaid);
  }

}
